pkg/cmd/admin/registry/registry.go

```diff
@@ -333,6 +239,14 @@ func RunCmdRegistry(f *clientcmd.Factory, cmd *cobra.Command, out io.Writer, cfg
 			}
 		}
 
+		// if we're using a daemon set we'll add it here and remove the DC.
+		// We allow the dc to be created above to get the goodness of app.AddServices
+		// which only supports DCs.
+		if cfg.DaemonSet {
+			daemonSet := generateDaemonSet(podTemplate, name, label)
+			objects = replaceDeploymentConfig(objects, daemonSet)
```

Comment:
Can't we just append desired object in both cases instead of replacing unwanted?